app/forms/provider_interface/offer_wizard.rb

```diff
@@ -6,28 +6,51 @@ class OfferWizard
               change_offer: %i[select_option providers courses study_modes locations conditions check] }.freeze
 
     attr_accessor :provider_id, :course_id, :course_option_id, :study_mode, :location_id,
-                  :standard_conditions, :further_condition_1, :further_condition_2,
-                  :further_condition_3, :further_condition_4, :current_step, :decision,
+                  :standard_conditions, :further_conditions, :current_step, :decision,
                   :action, :path_history, :wizard_path_history,
                   :provider_user_id, :application_choice_id
 
     validates :decision, presence: true, on: %i[select_option]
     validates :course_option_id, presence: true, on: %i[locations save]
     validates :study_mode, presence: true, on: %i[study_modes save]
     validates :course_id, presence: true, on: %i[courses save]
-    validates :further_condition_1, :further_condition_2, :further_condition_3, :further_condition_4, length: { maximum: 255 }
+    validate :further_conditions_valid, on: %i[conditions]
     validate :course_option_details, if: :course_option_id, on: :save
 
     def initialize(state_store, attrs = {})
       @state_store = state_store
 
       super(last_saved_state.deep_merge(attrs))
+      setup_further_conditions
       update_path_history(attrs)
     end
 
+    def self.build_from_application_choice(state_store, application_choice, options = {})
+      course_option = application_choice.offered_course_option || application_choice.course_option
+
+      conditions = (application_choice.offer || {})['conditions'].presence || MakeAnOffer::STANDARD_CONDITIONS
+
+      further_conditions = (conditions - MakeAnOffer::STANDARD_CONDITIONS).map.with_index.to_h do |c, index|
```

Comment:
Can this be extracted into its own method and more meaninful variable names used, e.g. condition instead of c.